jamon-parent/jamon/pom.xml

```diff
@@ -20,7 +20,7 @@
 					<version>${maven-surefire-plugin.version}</version>
 					<configuration>
 						<!-- https://github.com/wicketstuff/core/issues/676 -->
-						<reuseForks>false</reuseForks>
+						<forkedProcessExitTimeoutInSeconds>200</forkedProcessExitTimeoutInSeconds>
```

Comment:
IMO we should find the problem in your setup. Not hide it under the carpet.

Reply:
I could easily reproduce the exact same problem on a different Windows 10 1809 using the same Java and Maven. That system is somewhat slower than mine and reliably fails with `false` for `reuseForks` already. Replacing that with `forkedProcessExitTimeoutInSeconds` and `200` instead keeps the test failing, but it runs far longer until it fails. Setting a value of `500` makes the test immediately succeed:

[WARNING] Tests run: 7, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 120.519 s - in org.wicketstuff.jamon.request.cycle.JamonMonitoredRequestCycleTest
	[INFO]
	[INFO] Results:
	[INFO]
	[WARNING] Tests run: 24, Failures: 0, Errors: 0, Skipped: 2
	[INFO]
	[INFO] ------------------------------------------------------------------------
	[INFO] BUILD SUCCESS
	[INFO] ------------------------------------------------------------------------
	[INFO] Total time:  05:22 min
	[INFO] Finished at: 2019-10-03T14:33:33+02:00
	[INFO] ------------------------------------------------------------------------

Reply:
Might the root cause of this problem be the used monitors? Those are background threads, so `forkedProcessExitTimeoutInSeconds` influences them and those within the tests itself seem to never be stopped:

    assertEquals(0, MonitorFactory.getMonitor("JamonAdminPage", "ms.").getHits(), 0);

I don't see how resetting the factory disabled aquired monitors:

https://github.com/stevensouza/jamonapi/blob/master/jamon/src/main/java/com/jamonapi/MonitorFactory.java#L335

I get a lot of output on STDOUT during running the tests and maybe that output is simply breaking the goodbye message Surefire expects.

Reply:
Same problem on Windows 7 with same version of Java and Maven:

	[INFO] ------------------------------------------------------------------------
	[INFO] BUILD SUCCESS
	[INFO] ------------------------------------------------------------------------
	[INFO] Total time:  04:26 min
	[INFO] Finished at: 2019-10-06T13:37:18+02:00
	[INFO] ------------------------------------------------------------------------